test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -192,6 +261,1427 @@ public void testCreateDeleteKeyValueTable() {
+
+        assertEquals("Keys should be same for both entries",
+                tableEntry1.getKey().getKey(), tableEntry2.getKey().getKey());
+        assertEquals("Corresponding value for entry1 should be as inserted",
+                value1, tableEntry1.getValue());
+        assertEquals("Corresponding value for entry2 should be as inserted",
+                value2, tableEntry2.getValue());
+        assertNotEquals("Corresponding values should be different for both the entries",
+                tableEntry1.getValue(), tableEntry2.getValue());
+
+        log.info("Table entry1, key: {} and value: {}", tableEntry1.getKey(), tableEntry1.getValue());
+        log.info("Table entry2, key: {} and value: {}", tableEntry2.getKey(), tableEntry2.getValue());
+        log.info("Successfully completed test Insert Same Key into multiple KeyFamilies");
+    }
+
+    /**
+     * Test case-43: Conditional insert KVP, if key not exists already
+     */
+    @Test
+    public void testInsertIfExists() {
+        KeyValueTableInfo newKVT = newKeyValueTableName();
+        KeyValueTable<Integer, String> newKeyValueTable = getKeyValueTable(newKVT);
+
+        Integer key = getKeyID();
+        String value = getValue();
+
+        Version version1 = newKeyValueTable.putIfAbsent(null, key, value).join();
+        assertNotNull("Vesrion1 should not be null", version1);
+        log.info("version1: {}", version1);
+        TableEntry<Integer, String> tableEntry1 = newKeyValueTable.get(null, key).join();
+        assertEquals("Corresponding key for entry1 should be as inserted",
+                key, tableEntry1.getKey().getKey());
+        log.info("Get tableEntry1: {}", tableEntry1);
+        assertEquals("Corresponding value for entry1 should be as inserted",
+                value, tableEntry1.getValue());
+
+        AssertExtensions.assertSuppliedFutureThrows(
+                "putIfAbsent did not throw Exception for already existing key",
+                () -> newKeyValueTable.putIfAbsent(null, key, value),
+                ex -> ex instanceof BadKeyVersionException
+        );
+        log.info("Found BadKeyVersionException for putIfAbsent insert for already existing key");
+
+        List<Integer> searchKey = new ArrayList<Integer>();
+        searchKey.add(key);
+        List<TableEntry<Integer, String>> getAllEntry = newKeyValueTable.getAll(null, searchKey).join();
+        assertEquals("Only one entry should be present from the fetched list with the given key",
+                1, getAllEntry.size());
+        log.info("Only one entry is present from the Fetched list of 'if-exists-test': {}", getAllEntry);
+        log.info("Successfully completed test Conditional insert KVP, if key not exists already");
+    }
+
+    /**
+     * Test case-44: Conditional update single TableEntry value using its version
+     */
+    @Test
+    public void testConditionUpdateKeyValue() {
```

Comment:
This is already covered in the base class.

Reply:
Sure, I'll remove this test as its a duplicate.